tools/analyzer_plugin/lib/src/async_plugin_apis/diagnostic.dart

```diff
@@ -68,21 +69,15 @@ mixin DiagnosticMixin on ServerPlugin {
     final analysisOptions = pluginOptionsReader.getOptionsForResult(analysisResult);
 
     try {
-      // If there is no relevant analysis result, notify the analyzer of no errors.
-      if (analysisResult.unit == null) {
-        channel.sendNotification(plugin.AnalysisErrorsParams(analysisResult.path!, []).toNotification());
-        return [];
-      }
-
```

Comment:
I assume this was removed because analysisResult.unit is non-nullable?

Reply:
Yup, exactly.